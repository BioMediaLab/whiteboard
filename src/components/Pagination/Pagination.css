.Pagination {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
}
