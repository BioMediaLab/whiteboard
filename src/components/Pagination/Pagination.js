import React from 'react'
import Button from '../Button'
import './Pagination.css'

export default function Pagination({
  onFirstClick,
  onPreviousClick,
  onNextClick,
  onLastClick
}) {
  return (
    <div className="Pagination">
      <div className="PaginationPrevious">
        <Button onClick={onFirstClick}>First</Button>
        <Button onClick={onPreviousClick}>Previous</Button>
      </div>
      <div className="PaginationCenter">Viewing page 1 of 20</div>
      <div className="PaginationNext">
        <Button onClick={onNextClick}>Next</Button>
        <Button onClick={onLastClick}>Last</Button>
      </div>
    </div>
  )
}
