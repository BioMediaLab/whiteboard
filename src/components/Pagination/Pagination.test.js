import React from 'react'
import Pagination from './'

describe('Pagination', () => {
  test('can render', () => {
    const component = render(<Pagination />)

    expect(component).toMatchSnapshot()
  })

  test('can handle button clicks', () => {
    const onFirstClick = jest.fn()
    const onLastClick = jest.fn()
    const onNextClick = jest.fn()
    const onPreviousClick = jest.fn()
    const component = render(
      <Pagination
        onFirstClick={onFirstClick}
        onLastClick={onLastClick}
        onNextClick={onNextClick}
        onPreviousClick={onPreviousClick}
      />
    )

    component.clickText('First')
    component.clickText('Last')
    component.clickText('Next')
    component.clickText('Previous')

    expect(onFirstClick).toHaveBeenCalled()
    expect(onLastClick).toHaveBeenCalled()
    expect(onNextClick).toHaveBeenCalled()
    expect(onPreviousClick).toHaveBeenCalled()
  })
})
