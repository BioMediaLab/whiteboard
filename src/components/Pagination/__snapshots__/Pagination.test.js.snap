// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination can render 1`] = `
<div>
  <div
    class="Pagination"
  >
    <div
      class="PaginationPrevious"
    >
      <button
        class="Button"
        type="button"
      >
        First
      </button>
      <button
        class="Button"
        type="button"
      >
        Previous
      </button>
    </div>
    <div
      class="PaginationCenter"
    >
      Viewing page 1 of 20
    </div>
    <div
      class="PaginationNext"
    >
      <button
        class="Button"
        type="button"
      >
        Next
      </button>
      <button
        class="Button"
        type="button"
      >
        Last
      </button>
    </div>
  </div>
</div>
`;
